akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/Effect.scala

```diff
@@ -203,6 +203,33 @@ object Effect {
     def duration(): java.time.Duration = delay.asJava
   }
 
+  final case class TimerScheduled[U](
+      key: Any,
+      msg: U,
+      delay: FiniteDuration,
+      mode: TimerScheduled.TimerMode,
+      overriding: Boolean)(val send: () => Unit)
+      extends Effect {
+    def duration(): java.time.Duration = delay.asJava
+  }
+
+  object TimerScheduled {
+    sealed trait TimerMode
+    case object FixedRateMode extends TimerMode
+    case object FixedDelayMode extends TimerMode
+    case object SingleMode extends TimerMode
+
+    /*Java API*/
+    def fixedRateMode = FixedRateMode
+    def fixedDelayMode = FixedDelayMode
+    def singleMode = SingleMode
+  }
+
+  /*Java API*/
+  def timerScheduled = TimerScheduled
```

Comment:
I think this may have a weird return type in Java, so probably slightly better to either move the `TimerScheduled` out of `Effect` or do the accessor methods directly on `Effect`.

Reply:
On second thought, this is fine. Let it be as is.

Reply:
according to Intelij idea:
```
Effect.TimerScheduled$ ts = Effect.timerScheduled();
```

I can extract a trait for `TimerScheduledCompanion` and expose that for the java api (and of course derive it for the scala API's companion object)